js/typeahead.js

```diff
@@ -118,16 +159,22 @@ function Typeahead(selector, type, url) {
   this.url = url || '/';
   this.typeahead = null;
 
-  this.init(type || 'candidates');
+  // If there's a type defined, use that dataset; otherwise use all of them
+  if (type) {
+    this.dataset = datasets[type];
+  } else {
+    this.dataset = Object.values(datasets);
+  }
+
+  this.init();
 
   events.on('searchTypeChanged', this.handleChangeEvent.bind(this));
 }
 
-Typeahead.prototype.init = function(type) {
+Typeahead.prototype.init = function() {
```

Comment:
Just noting here that changing the function signature will have an impact on anything calling it if it's not also updated.  In this case it's not an issue since any call still passing a value to this method won't be impacted as the method will just ignore the value, but if it were the opposite case (adding an argument), we'd want to be sure we updated all of the calls accordingly to match.